core/src/main/java/com/datastrato/gravitino/storage/kv/RocksDBOptions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.storage.kv;
+
+import com.datastrato.gravitino.Config;
+import org.rocksdb.WriteOptions;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.Options;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import lombok.Getter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RocksDBOptions {
+    public static final Logger LOGGER = LoggerFactory.getLogger(RocksDBKvBackend.class);
+    @Getter
+    private final Map<String, BiConsumer<RocksDBOptions, String>> optionSetters;
+    @Getter
+    private Options options;
+
+    @Getter
+    private WriteOptions writeOptions;
+
+    @Getter
+    private ReadOptions readOptions;
+
+    public RocksDBOptions() {
+        this.options = new Options();
+        this.writeOptions = new WriteOptions();
+        this.readOptions = new ReadOptions();
+        this.optionSetters = new HashMap<>();
+        initializeOptionSetters();
+    }
+
+    public RocksDBOptions(Options options, WriteOptions writeOptions, ReadOptions readOptions) {
+        this.options = options;
+        this.writeOptions = writeOptions;
+        this.readOptions = readOptions;
+        this.optionSetters = new HashMap<>();
+        initializeOptionSetters();
+    }
+
+    private void initializeOptionSetters() {
+        // Each option name maps to a lambda that applies the setting to the appropriate
+        // option object
+        optionSetters.put("gravitino.entity.store.kv.rocksdb.options.maxBackgroundJobs",
```

Comment:
I believe you can shorten it to `options.maxBackgroundJobs` or `maxBackgroundJobs`.